File: Python/Gestion de Disques/6.2.py
import os   # pour parcourir les dossiers et fichiers
import sys  # pour récupérer les arguments

# Vérifier que l'utilisateur fournit un répertoire
if len(sys.argv) != 2:
    print(f"Usage : {sys.argv[0]} <répertoire>")
    sys.exit(1)

dossier = sys.argv[1]

# Parcourir tous les fichiers du répertoire
for nom_fichier in os.listdir(dossier):
    chemin_complet = os.path.join(dossier, nom_fichier)  # chemin complet
    
    # Vérifier que c'est bien un fichier (ignorer les dossiers)
    if os.path.isfile(chemin_complet):
        taille = os.path.getsize(chemin_complet)  # taille en octets
        uid = os.stat(chemin_complet).st_uid     # propriétaire (UID)
        
        # Afficher uniquement les fichiers > 1 Mo
        if taille > 1024*1024:
            print(f"Fichier : {nom_fichier}, UID={uid}, taille={taille} octets")

File: C/Gestion de Disques/6.2.c
#include <stdio.h>
#include <dirent.h>  // pour DIR et readdir
#include <sys/stat.h> // pour struct stat et stat()
#include <string.h>  // pour strcat, strcpy
#include <stdlib.h>  // pour exit

int main(int argc, char *argv[])
{
    if(argc != 2) {
        printf("Usage : %s <répertoire>\n", argv[0]);
        return 1;
    }

    DIR *d = opendir(argv[1]); // ouvrir le répertoire passé en argument
    if(!d) {
        perror("Erreur ouverture dossier");
        return 1;
    }

    struct dirent *file;

    // parcourir tous les fichiers du dossier
    while((file = readdir(d)) != NULL)
    {
        struct stat st;
        char path[1024];

        // construire le chemin complet : dossier + nom du fichier
        snprintf(path, sizeof(path), "%s/%s", argv[1], file->d_name);

        if(stat(path, &st) == -1) { // lire les infos du fichier
            perror("Erreur stat");
            continue;
        }

        // afficher uniquement les fichiers > 1 Mo
        if(st.st_size > 1024*1024) {
            printf("Fichier : %s, UID=%d, taille=%ld octets\n",
                   file->d_name, st.st_uid, st.st_size);
        }
    }

    closedir(d); // fermer le répertoire
    return 0;
}